lib/lte_link_control/lte_lc.c

```diff
@@ -151,32 +182,328 @@ static const char cgauth[] = "AT+CGAUTH="CONFIG_LTE_PDN_AUTH;
 static const char legacy_pco[] = "AT%XEPCO=0";
 #endif
 
-void at_handler(void *context, const char *response)
+enum lte_lc_notif_type {
+	LTE_LC_NOTIF_CEREG,
+	LTE_LC_NOTIF_CSCON,
+	LTE_LC_NOTIF_CEDRXP,
+
+	LTE_LC_NOTIF_COUNT,
+};
+
+static const char *const at_notifs[] = {
+	[LTE_LC_NOTIF_CEREG] = "+CEREG",
+	[LTE_LC_NOTIF_CSCON] = "+CSCON",
+	[LTE_LC_NOTIF_CEDRXP] = "+CEDRXP",
+};
+
+BUILD_ASSERT(ARRAY_SIZE(at_notifs) == LTE_LC_NOTIF_COUNT);
+
+static bool is_relevant_notif(const char *notif, enum lte_lc_notif_type *type)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(at_notifs); i++) {
+		if (strncmp(at_notifs[i], notif,
+			    strlen(at_notifs[i])) == 0) {
+			/* The notification type matches the array index */
+			*type = i;
+
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static int parse_cereg(const char *notification,
+		       enum lte_lc_nw_reg_status *reg_status,
+		       struct lte_lc_cell *cell,
+		       struct lte_lc_psm_cfg *psm_cfg)
+{
+	int err, status;
+	struct at_param_list resp_list = {0};
+	char str_buf[10];
+	size_t len = sizeof(str_buf) - 1;
+
+	if (notification == NULL) {
+		return -EINVAL;
+	}
+
+	err = at_params_list_init(&resp_list, AT_CEREG_PARAMS_COUNT_MAX);
+	if (err) {
+		LOG_ERR("Could not init AT params list, error: %d", err);
+		return err;
+	}
+
+	/* Parse CEREG response and populate AT parameter list */
+	err = at_parser_max_params_from_str(notification,
```

Comment:
The max number of parameters is the same value used in `at_params_list_init`, so you do not need to specify it. The last parameter is only useful when the allocated list is bigger than the number of parameters you want to parse.

Reply:
Thanks, classic copy/paste without thinking... I'm switching to use `at_parser_params_from_str()` instead